node/api/host.go

```diff
@@ -61,6 +67,44 @@ func folderIndex(folderPath string, storageFolders []modules.StorageFolderMetada
 	return -1, errStorageFolderNotFound
 }
 
+// hostContractInfoHandler handles the API call to get the contract information of the host.
+// Information is retrieved via the storage obligations from the host database.
+// TODO: filters are hard coded. Adding or removing storage obligation statuses will
+// currently break the API.
+func (api *API) hostContractInfoHandler(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	statusStr := req.FormValue("status")
+	if statusStr == "" {
+		WriteError(w, Error{"Status must be provided to a /host/contracts call."}, http.StatusBadRequest)
+		return
+	}
+	var filter uint64
+	switch statusStr {
+	case "unresolved":
+		filter = 0
+	case "rejected":
+		filter = 1
+	case "succeeded":
+		filter = 2
+	case "failed":
+		filter = 3
+	case "all":
+		filter = 4
+	default:
+		WriteError(w, Error{"Unable to parse contract status."}, http.StatusBadRequest)
+		return
+	}
+	sos := []modules.StorageObligation{}
+	for _, so := range api.host.StorageObligations() {
+		if so.ObligationStatus == filter || filter == 4 {
+			sos = append(sos, so)
```

Comment:
I'm wondering if it wouldn't be simpler to just return all of the obligations, and let the caller filter them.
The tradeoff, of course, is that you may end up sending a ton of obligations when the caller only cares about a few. But as long as it's not a prohibitively large amount (i.e. thousands of large JSON objects), it's likely not going to be an issue. Also, if it does become an issue, we'd probably want to making the API paginated rather than filtering based on status. Consider that, even if we do filter based on status, we'd still end up needing pagination in the long run anyway, because a status like `all` is going to return too much data.

Reply:
I agree it is simpler to just return all the obligations. However, since the ObligationStatus is an `int`, I'm not sure how the caller would know the "real" status of the obligations. That's why I chose the named filter where it is clear for the caller what he receives.

Reply:
It doesn't look like the `modules.StorageObligation` struct is being used anywhere except tests, so it's not bound by compatibility requirements. That means we can change the `ObligationStatus` field to a `string` (or maybe a `types.Specifier`?). Unless @DavidVorick has objections, I think this is the right approach to take. Something like:
```go
// modules/host.go

const (
    ObligationStatusSucceeded = "succeeded"    
    ObligationStatusRejected  = "rejected"
    // ... etc
)

type StorageObligation struct {
    // ... etc
    ObligationStatus string `json:"obligationstatus"`
}
```

Reply:
@lukechampine implementing your review comments and I would like to use a switch statement in the `StorageObligations` function but I'm not sure how to throw an error when there is no match:
```go
// modules/host/storageobligations.go

switch so.StorageObligationStatus {
case obligationSucceeded:
    varString = modules.ObligationStatusSucceeded
case obligationRejected:
    varString = modules.ObligationStatusRejected
// ... etc
default:
    // how to error here, use return build.Err?
}
```
any thoughts on this?

Reply:
if there's no match, the data should be treated as corrupted. I would call `h.log.Severe("WARN: invalid obligation status")` (with other helpful info included) and then skip the obligation entirely, moving on top the next one.

Also, a map may be more readable than a switch:
```go
varString := map[uint64]string{
    obligationSucceeded: modules.ObligationStatusSucceeded,
    obligationRejected:  modules.ObligationStatusRejected,
}[so.StorageObligationStatus]
if varString == "" {
    // handle invalid status
}
```